src/sentry/api/endpoints/release_threshold.py

```diff
@@ -84,18 +84,16 @@ def get(self, request: Request, project: Project) -> HttpResponse:
         )
         if not serializer.is_valid():
             return Response(serializer.errors, status=400)
+
+        release_thresholds = ReleaseThreshold.objects.filter(project=project)
         environment_name: Union[str, None] = request.GET.get("environment")
-        if not environment_name:
-            release_thresholds = ReleaseThreshold.objects.filter(project=project)
-            return Response(serialize(list(release_thresholds), request.user), status=200)
-        else:
+
+        if environment_name:
             try:
                 environment = Environment.objects.get(
                     name=environment_name, organization_id=project.organization_id
                 )
             except Environment.DoesNotExist:
                 raise Http404
-            release_thresholds = ReleaseThreshold.objects.filter(
-                project=project, environment=environment
-            )
+            release_thresholds = release_thresholds.filter(environment=environment)
```

Comment:
Oh 🤔 

I guess this is fine
I was thinking more along the lines of 
```
from django.db.models import Q

query = Q(
    project=project
)
if environment_name:
    q &= Q(environment=environment_name)
```

Reply:
i think something like this works. also since you pass in the name of the environment you don't actually have to fetch the environment object -- you can do `filter(environment__name=environment_name)`